ui/pages/send/send-content/send-hex-data-row/send-hex-data-row.component.js

```diff
@@ -13,16 +13,36 @@ export default class SendHexDataRow extends Component {
     t: PropTypes.func,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      hexValue: null,
+    };
+  }
+
   onInput = (event) => {
     const { updateSendHexData } = this.props;
     const data = event.target.value.replace(/\n/gu, '') || null;
     updateSendHexData(data);
   };
 
+  componentDidMount() {
+    this.setState({ hexValue: this.props.data });
+  }
+
+  componentDidUpdate(prevProps) {
+    const { data: prevPropsHexValue } = prevProps;
+    const { data: propsHexValue } = this.props;
+
+    if (prevPropsHexValue !== propsHexValue) {
+      this.setState({ hexValue: propsHexValue });
```

Comment:
In react , `props` are indeed intended to be read-only. That’s not possible to change the value of `props` directly within a react component . If you need to modify a value based on the `props`, you should create and manage `state` within the component.

the error we've had is due to the fact that we're trying to change the value of the `props` directly, so as long as we don't refresh we're always left with the value of the previous `props` which doesn't change.

the solution was to manage this through states within the component

Reply:
I'm not following. Where were we trying to change props directly in the old file? It looks like we were just passing data from props into the SendRowWrapper, whereas now we are saving the new value in state? 

Also just as an aside i'd prefer we convert this to a stateless functional component instead of adding deprecated methods to the component making it more difficult to upgrade the codebase in the future

Reply:
hey @brad-decker ,
i have he same idea to convert it to a functional component , but i thought it will be out of scope.
but i'll do it now.
the data are coming from the `props` ( check the data object ), then we are using it to init our state , when you click the edit button we are calling the function `updateUserInputHexData` with empty string ( file `/ui/ducks/send/send.js` line 2082 ) to change the state but this one took an immutable value ( initialised using props ) and that's why you'll always have the hex data.

so the bug was coming from the fact that the function `updateUserInputHexData` called when we trigger the edit button and switch token who was supposed to set the input as empty

Reply:
The previous version of this component did not have local state, it used redux state. This is where I am getting confused because your PR is adding a local shallow copy of the redux state into the state of this component.

Reply:
I'll try to find out why the call to the function `updateUserInputHexData` is not setting the redux state to an empty string.
my assumption was , because this information have the same memory reference than the props.
i'm keen to convert it to functional component as well , i like the idea.
i'll ping you when i'll do all the changes , thnx for your comment @brad-decker  :)

Reply:
hey @brad-decker ,
i was able to find the origin of the error and fix it , there's no need to change any thing in the component level.
thnx for your comment this was helpful to go deeper and find a best way to fix it